packages/backend-defaults/src/entrypoints/database/connectors/postgres.ts

```diff
@@ -77,44 +78,48 @@ export async function buildPgDatabaseConfig(
     overrides,
   );
 
-  if (config.connection?.type === 'cloudsql') {
-    if (config.client !== 'pg') {
-      throw new Error('Cloud SQL only supports the pg client');
-    }
-
-    if (!config.connection.instance) {
-      throw new Error('Missing instance connection name for Cloud SQL');
-    }
+  let transformedConfig = config;
 
-    const {
-      Connector: CloudSqlConnector,
-      IpAddressTypes,
-      AuthTypes,
-    } = await import('@google-cloud/cloud-sql-connector');
-    const connector = new CloudSqlConnector();
-    const clientOpts = await connector.getOptions({
-      instanceConnectionName: config.connection.instance,
-      ipType: IpAddressTypes.PUBLIC,
-      authType: AuthTypes.IAM,
-    });
-
-    const cloudsqlConfig = {
-      ...config,
-      client: 'pg',
-      connection: {
-        ...config.connection,
-        ...clientOpts,
-      },
-    };
+  if (config.connection.type) {
+    if (config.connection.type === 'cloudsql') {
+      if (config.client !== 'pg') {
+        throw new Error('Cloud SQL only supports the pg client');
+      }
 
-    // Trim additional properties from the connection object passed to knex
-    delete cloudsqlConfig.connection.type;
-    delete cloudsqlConfig.connection.instance;
+      if (!config.connection.instance) {
+        throw new Error('Missing instance connection name for Cloud SQL');
+      }
 
-    return cloudsqlConfig;
+      const {
+        Connector: CloudSqlConnector,
+        IpAddressTypes,
+        AuthTypes,
+      } = await import('@google-cloud/cloud-sql-connector');
+      const connector = new CloudSqlConnector();
+      const clientOpts = await connector.getOptions({
+        instanceConnectionName: config.connection.instance,
+        ipType: IpAddressTypes.PUBLIC,
+        authType: AuthTypes.IAM,
+      });
+
+      transformedConfig = {
+        ...config,
+        client: 'pg',
+        connection: {
+          ...config.connection,
+          ...clientOpts,
+        },
+      };
+    } else if (config.connection.type !== 'default') {
+      throw new Error(`Unknown connection type: ${config.connection.type}`);
+    }
   }
 
-  return config;
+  // Remove the connection type and instance from the config
```

Comment:
```suggestion
  // Trim additional properties from the connection object passed to knex
```

original comment was better